From f2068f24900f1a3afd6aae72e9739ce74ef03fd8 Mon Sep 17 00:00:00 2001
From: Zhen Han <zhen.han@intel.com>
Date: Wed, 26 Dec 2018 10:52:13 +0800
Subject: [PATCH 0958/1214] i915: Set initial panel_power_off_time to 0

It requires eDp panel is fully powered off before next power-on.
Specifically, in function of edp_panel_vdd_on(), it needs to
wait t11_t12 after LCDVCC is is off. Usually, the t12 is 500ms.
That is it needs to wait the time of "t11_t12 - power_off_duration)" if
the plane is off. Here: power_off_duration = (now_time - last_off_timestamp).

However, currently, this value is set to i915 module initial timestamp.
It is not the real last_off_timestamp and makes i915 driver
waiting more time.

static void intel_dp_init_panel_power_timestamps(struct intel_dp
*intel_dp)
{
	intel_dp->panel_power_off_time = ktime_get_boottime();
	intel_dp->last_power_on = jiffies;
	intel_dp->last_backlight_off = jiffies;
}

This patch sets "intel_dp->panel_power_off_time" as 0 which could save
~200ms in cold boot time.

Change-Id: I9ec340bbc7e3736fa62b57ac8762bfac45437bf3
Tracked-On: OAM-74882
Tracked-On: OAM-74847
Signed-off-by: Zhang Ning <ning.a.zhang@intel.com>
Reviewed-by: Zhang, Baoli <baoli.zhang@intel.com>
---
 drivers/gpu/drm/i915/intel_dp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index 9474585..345de02 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -5418,7 +5418,7 @@ intel_dp_add_properties(struct intel_dp *intel_dp, struct drm_connector *connect
 
 static void intel_dp_init_panel_power_timestamps(struct intel_dp *intel_dp)
 {
-	intel_dp->panel_power_off_time = ktime_get_boottime();
+	intel_dp->panel_power_off_time = ktime_set(0, 0);
 	intel_dp->last_power_on = jiffies;
 	intel_dp->last_backlight_off = jiffies;
 }
-- 
2.7.4

